c7n/resources/iam.py

```diff
@@ -159,6 +145,34 @@ def process(self, resources, event=None):
         return results
 
 
+@Role.filter_registry.register('inline-policy-used')
+class IamRoleInlinePolicyUsed(Filter):
+
+    schema = type_schema('inline-policy-used')
+
+    def _inline_policies(self, resource):
+        client = local_session(self.manager.session_factory).client('iam')
```

Comment:
preference would be to have client passed into the function, constructing clients is expensive, so construction in a loop is best avoided.